Compatibility of Monoidal and (Co)Cartesian Structure

arthurovale opened the pull request:
Several interactions between structural morphisms of monoidal structure and cocartesian structure.

jeremie-koenig reviewed (commented):
Thanks Arthur! A few nitpicks below.

Please make sure your new additions are documented and formatted consistently with the rest of the file:
  * The rest of the module uses `Proposition` as the "default" proof keyword, you should do this for your properties as well (that may not be the most sensible choice but that's a separate question; we should be consistent at least within the same file at least until we codify a particular convention)
  * Use vertical and horizontal indentation consistent with the rest of the file.
  * Same for spacing around `:`

Make sure you use proper *coqdoc* sectioning and formatting:
  * coqdoc primer [here](https://rocq-prover.org/doc/V8.18.0/refman/using/tools/coqdoc.html)
  * generated doc for the library can be browsed [here](https://certikos.github.io/rbgs/html/toc.html), or use `make html` locally
  * The properties above the ones you added are in their own "Monoidal Structure" subsection and should be grouped since they implement the relevant part of the SymmetricMonoidalStructureDefinition interface. You should start a new subsection of the appropriate depth here.

The symmetry between the cartesian/cocartesian should be preserved, so if you add these properties to one you should add the corresponding properties to the other.

Ideally, please use a commit message that will make sense once it's incorporated into the commit history (so i can use your commit as-is and keep the right attribution, etc).

Less critical but I'm also curious about your goal in adding these properties. Do you need them for a particular application? How did they come up? They make a lot of sense to me as generic things to have about the way the various operations interact with each other, but I'm wondering if there is some consistent scheme we could use to decide what to include or not: can we build some rewriting system that has some good normalization property? etc.

Anyway thanks again for the contribution and let me know if you have any thoughts / questions / etc.

arthurovale committed:
Add html/ to .gitignore

arthurovale committed:
Compatibility of Monoidal and (Co)Cartesian Structure

Added naturality of projections/injections and compatibility
of projections/injections with the monoidal structural
morphisms. This should avoid unfolding structural morphisms
when reasoning about the (co)cartesian structure.
